# Урок 5

## Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование.

In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
from scipy import stats
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

### Задача 1

#### Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

In [2]:
n, m0, scale = 100, 17.5, 2
mu = 17
sigma = scale / np.sqrt(n)
alpha = 0.025 

In [3]:
(mu + 2 * sigma, float('inf'))

(17.4, inf)

Среднее значение выборки из 100 входит в правостороннюю критическую область 17.5 > 17.4, поэтому нулевая гипотеза, что шарики для подшипников имеют средний диметр 17 мм отвергается в пользу альтернативной гипотезы, что средний димаетр не равен 17 мм.

### Задача 2

#### Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания a с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

In [4]:
scale, m, n = 16, 80, 256
print('Искомый интервал:',(m - 2 * scale / np.sqrt(n), m + 2 * scale / np.sqrt(n)))

Искомый интервал: (78.0, 82.0)


### Задача 3

#### Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

#### $202, 203, 199, 197, 195, 201, 200, 204, 194, 190$

#### Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [5]:
sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean = sample.mean()
std = sample.std(ddof=1)
n = 10
alpha = 0.01
t = stats.t.ppf(1 - alpha / 2, df=n - 1)
print('t=',np.round(t,3))
print('Доверительный интервал:',np.round((mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n)),2))

t= 3.25
Доверительный интервал: [193.92 203.08]


Утверждение продавца верно.